File: hackerrank/IsFibo.java
// https://www.hackerrank.com/challenges/is-fibo

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {


    
 public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int _cases;
        _cases = in.nextInt();
     
        long n;
     
        for(int i=0; i<_cases; i++) {
            
            n = in.nextLong();

            if(Math.sqrt(5*Math.pow(n,2) + 4) % 1 == 0 || Math.sqrt(5*Math.pow(n,2) - 4) %1 == 0)
                System.out.println("IsFibo");
            else
                System.out.println("IsNotFibo");
        }     
   }
}

File: hackerrank/IsFibo.cpp
// https://www.hackerrank.com/challenges/is-fibo

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


bool isFibo(long a) {
  return  (fmod((sqrt(5*pow(a,2) + 4)),1) == 0) || (fmod((sqrt(5*pow(a,2) - 4)),1) == 0);
}

int main() {
  int _cycles;
  long number;
  cin>>_cycles;
    
  for(int i=0;i<_cycles;i++) {
    cin>>number;
    if(isFibo(number)) {
          cout<<"IsFibo";
    }
    else {
          cout<<"IsNotFibo";
    }
   cout<<'\n';
  }
    
  return 0;
}